obspy/core/utcdatetime.py

```diff
@@ -383,8 +396,36 @@ def __init__(self, *args, **kwargs):
             kwargs['second'] = int(_sec)
             kwargs['microsecond'] = int(round(_frac * 1e6))
             args = args[0:5]
-        dt = datetime.datetime(*args, **kwargs)
-        self._from_datetime(dt)
+
+        try:  # If a value Error is raised try to allow overflow (see #2222)
+            dt = datetime.datetime(*args, **kwargs)
+        except ValueError:
+            if not strict:
+                self._handle_overflow(*args, **kwargs)
+            else:
+                raise
+        else:
+            self._from_datetime(dt)
+
+    def _handle_overflow(self, year, month, day, hour=0, minute=0, second=0,
+                         microsecond=0):
+        """
+        Handles setting date if an overflow of usual value limits is detected.
+        """
+        kwargs = locals()
+        kwargs.pop('self')
```

Comment:
I'm not at all comfortable with using `locals()` here. It can be utterly polluted with other stuff e.g. in debugger sessions etc. Plus there is no need at all for beeing magicky here, we can just be simple, explicit and stable.

Explicit is better than implicit. ;-)

```
kwargs = dict(
    year=year, month=month, day=day, hour=hour, minute=minute,
    second=second, microsecond=microsecond)
```

Or just put all of them variables in the UTCDateTime call at the end?

Reply:
Sure no problem.